crates/dynamodb-streams/src/stream_state.rs

```diff
@@ -0,0 +1,1963 @@
+use crate::checkpoint::{CheckpointPosition, GlobalCheckpoint, ShardCheckpoint};
+use crate::client_sdk::{ApiShard, SDKClient};
+use aws_sdk_dynamodbstreams::types::{Record, ShardIteratorType};
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::SystemTime;
+
+#[derive(Debug)]
+pub struct RecordBatch {
+    pub shard_id: String,
+    pub records: Vec<Record>,
+    pub checkpoint: ShardCheckpoint,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ActiveShard {
```

Comment:
Might be cleaner to do 
```rust
pub enum ShardState {
    Active {
        pub shard_id: String,
        pub parent_shard_id: Option<String>,
        pub iterator: String,
    },
    Pending {
        pub shard_id: String,
        pub parent_shard_id: Option<String>,
    },
    Initializing {
        pub shard_id: String,
        pub parent_shard_id: Option<String>,
    },
}
```
That way we don't need three HashMaps that we remove and reinsert when state transitions occur.

Reply:
e.g. `StreamState::reinitialize_shard`, `StreamState::mark_active`